src/20250814_Multi_GHOInkLaunch/abstraction/AaveV3GHOLane.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IProposalGenericExecutor} from 'aave-helpers/src/interfaces/IProposalGenericExecutor.sol';
+import {IUpgradeableBurnMintTokenPool_1_5_1} from 'src/interfaces/ccip/tokenPool/IUpgradeableBurnMintTokenPool.sol';
+import {IRateLimiter} from 'src/interfaces/ccip/IRateLimiter.sol';
+import {GhoCCIPChains} from './constants/GhoCCIPChains.sol';
+
+/**
+ * @title AaveV3GHOLane
+ * @author Aave Labs
+ * @notice Proposal Executor that sets Aave V3 GHO Lane through CCIP
+ */
+abstract contract AaveV3GHOLane is IProposalGenericExecutor {
+  IUpgradeableBurnMintTokenPool_1_5_1 public immutable TOKEN_POOL;
+
+  /**
+   * @notice Constructor
+   * @param localChainInfo The relevant information of the chain where the GHO CCIP Lanes are being configured
+   */
+  constructor(GhoCCIPChains.ChainInfo memory localChainInfo) {
+    TOKEN_POOL = IUpgradeableBurnMintTokenPool_1_5_1(localChainInfo.ghoCCIPTokenPool);
+  }
+
+  /**
+   * @notice Executes the GHO CCIP Lanes configuration proposal
+   */
+  function execute() external virtual {
+    _execute();
+  }
+
+  /**
+   * @notice Returns the GHO CCIP Lanes to add in the form of ChainUpdate's
+   * @return An array of ChainUpdate's with each GHO CCIP Lane that must be added via this proposal
+   */
+  function lanesToAdd()
+    public
+    view
+    virtual
+    returns (IUpgradeableBurnMintTokenPool_1_5_1.ChainUpdate[] memory);
+
+  /**
+   * @notice Returns the GHO CCIP Lanes to remove in the form of chain selectors. By default, no Lanes are removed
+   * @return An array of chain selectors of each of the GHO CCIP Lanes that must be removed via this proposal
+   */
+  function lanesToRemove() public view virtual returns (uint64[] memory) {
+    return new uint64[](0);
+  }
+
+  function _execute() internal virtual {
+    TOKEN_POOL.applyChainUpdates({
+      remoteChainSelectorsToRemove: lanesToRemove(),
+      chainsToAdd: lanesToAdd()
+    });
+  }
+
```

Comment:
could be good to have a method so that for normal launches of new lanes, where it has to comunicate with all existing lanes, it would automatically iterate through all the chains, without the need to manually add then with lanesToAdd.
This way we would just review "one" line, on the payload for the new chain, no need to check if all the lanes are added or not etc

Reply:
Makes sense. I added it and that also simplified a lot some test setups. The only drawback is that we need to maintain the list of "all chains" when adding a new one. But that's much simpler than always listing all chains manually. So I think it's a good trade-off.

The change was done here: https://github.com/bgd-labs/aave-proposals-v3/pull/802/commits/b17b079193a379877eee13d2ffd74588167213c1